usaspending_api/etl/es_transaction_template.json

```diff
@@ -160,72 +160,137 @@
-          "type": "keyword",
+          "type": "text",
           "fields": {
-            "raw": {
-              "type": "text"
+            "keyword": {
+              "type": "keyword"
             }
           }
         },
         "funding_toptier_agency_name": {
-          "type": "text"
+          "type": "text",
+          "fields": {
+            "keyword": {
+              "type": "keyword"
+            }
+          }
         },
         "awarding_subtier_agency_name": {
-          "type": "keyword",
+          "type": "text",
           "fields": {
-            "raw": {
-              "type": "text"
+            "keyword": {
+              "type": "keyword"
             }
           }
         },
         "funding_subtier_agency_name": {
-          "type": "text"
+          "type": "text",
+          "fields": {
+            "keyword": {
+              "type": "keyword"
+            }
+          }
         },
         "awarding_toptier_agency_abbreviation": {
-          "type": "text"
+          "type": "text",
+          "fields": {
+            "keyword": {
+              "type": "keyword"
+            }
+          }
         },
         "funding_toptier_agency_abbreviation": {
-          "type": "text"
+          "type": "text",
+          "fields": {
+            "keyword": {
+              "type": "keyword"
+            }
+          }
         },
         "awarding_subtier_agency_abbreviation": {
-          "type": "text"
+          "type": "text",
+          "fields": {
+            "keyword": {
+              "type": "keyword"
+            }
+          }
         },
         "funding_subtier_agency_abbreviation": {
-          "type": "text"
+          "type": "text",
+          "fields": {
+            "keyword": {
+              "type": "keyword"
+            }
+          }
+        },
+        "awarding_toptier_agency_code": {
+          "type": "keyword"
+        },
+        "funding_toptier_agency_code": {
+          "type": "keyword"
+        },
+        "awarding_subtier_agency_code": {
+          "type": "keyword"
+        },
+        "funding_subtier_agency_code": {
+          "type": "keyword"
+        },
+        "cfda_id": {
+          "type": "integer"
+        },
+        "cfda_number": {
+          "type": "keyword"
         },
-
         "cfda_title": {
           "type": "text",
-          "index": false
+          "fields": {
+            "keyword": {
+              "type": "keyword"
+            }
+          }
         },
         "cfda_popular_name": {
           "type": "text",
           "index": false
         },
         "type_of_contract_pricing": {
-          "type": "text",
-          "index": false
+          "type": "text"
         },
         "type_set_aside": {
-          "type": "text",
-          "index": false
+          "type": "text"
         },
         "extent_competed": {
-          "type": "text",
-          "index": false
+          "type": "text"
         },
         "pulled_from": {
```

Comment:
Is {{pulled_from}} still useful? Originally it was used to determine if the procurement transaction was for a contract or IDV (boolean "AWARD" or "IDV" values) in conjunction with null in {{type}}. After ES was implemented the {{type}} field is populated for IDVs with "IDV*"